src/test/java/de/retest/web/it/TestHealerCssSelectorIT.java

```diff
@@ -0,0 +1,113 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+
+import de.retest.web.AttributesUtil;
+import de.retest.web.selenium.AutocheckingRecheckDriver;
+
+class TestHealerCssSelectorIT {
+
+	private static WebDriver driver;
+
+	@BeforeAll
+	static void setup() throws MalformedURLException {
+		final ChromeOptions opts = new ChromeOptions();
+		opts.addArguments(
+				// Enable headless mode for faster execution.
+				"--headless",
+				// Use Chrome in container-based Travis CI enviroment (see https://docs.travis-ci.com/user/chrome#Sandboxing).
+				"--no-sandbox",
+				// Fix window size for stable results.
+				"--window-size=1200,800" );
+		//		final String page = "TestHealerCssSelectorIT.html";
+		final String page = "TestHealerCssSelectorIT-changed.html";
+		driver = new AutocheckingRecheckDriver( new ChromeDriver( opts ) );
+		driver.navigate().to( TestHealerCssSelectorIT.class.getResource( page ).toExternalForm() );
+		driver.findElement( By.id( "visited" ) ).click();
+	}
+
+	@MethodSource( "attributeSelectors" )
+	@ParameterizedTest( name = "CSS Selector: {0} expected html element ID: {1}" )
+	void should_find_elements_selected_by_CSS_attribute_selectors( final String selector, final String id ) {
+		assertThat( driver.findElement( By.cssSelector( selector ) ).getAttribute( AttributesUtil.ID ) )
+				.isEqualTo( id );
+	}
+
+	private static Stream<Arguments> attributeSelectors() {
+		return Stream.of( //
+				Arguments.of( ".checkable[checked]", "checked" ), //
+				Arguments.of( ".checkable[id=checked]", "checked" ), //
+				Arguments.of( "[value~=selector]", "attribute-word-selector" ), //
+				Arguments.of( "[value|=attribute]", "attribute-word-selector" ), //
+				Arguments.of( ".checkable[id^=\"checked\"]", "checked" ), //
+				Arguments.of( ".checkable[id$=\"nate\"]", "indeterminate" ), //
+				Arguments.of( ".checkable[id*=\"nche\"]", "unchecked" ) //
+		);
+	}
+
+	@MethodSource( "supportedPseudoClassSelectors" )
+	@ParameterizedTest( name = "Pseudo class {1}" )
+	void should_find_elements_selected_by_pseudo_class_using_css_selector( final String cssClass,
+			final String pseudoClass ) {
+		final By selector = By.cssSelector( "." + cssClass + ":" + pseudoClass );
+		assertThat( driver.findElement( selector ).getAttribute( AttributesUtil.ID ) ).isEqualTo( pseudoClass );
+	}
+
+	private static Stream<Arguments> supportedPseudoClassSelectors() {
+		return Stream.of( //
+				Arguments.of( "checkable", "checked" ), //
+				Arguments.of( "disable", "disabled" ), //
+				Arguments.of( "write-access", "read-only" ) //
+		);
+	}
+
+	@MethodSource( "unsupportedPseudoClassSelectors" )
+	@ParameterizedTest( name = "Pseudo class {1}" )
+	void should_not_find_elements_selected_by_unsupported_pseudo_class_using_css_selector( final String cssClass,

```

Comment:
I assume the test suite already is quite slow? Would it make sense to remove parts like this to a unit test?